pkg/handler/iam_handler.go

```
@@ -181,8 +180,7 @@ func updatePolicy(p *iampb.Policy, bs []*v1alpha1.Binding, expiry time.Time) err
 		// Skip expired bindings.
 		expired, err := expired(cb.Condition.Expression)
 		if err != nil {
-			// Return error immediately since we don't expect this to fail.
-			return fmt.Errorf("failed to check expiry: %w", err)
+			retErr = errors.Join(retErr, fmt.Errorf("failed to check expiry: %w", err))
```

This line particularly - we should ignore this error